api/app/Builders/UserBuilder.php

```diff
@@ -0,0 +1,567 @@
+     */
+    public function whereHasPoolCandidateCommunity(?string $communityId): self
+    {
+        if (empty($communityId)) {
+            return $this;
+        }
+
+        return $this->whereHas('poolCandidates', function ($query) use ($communityId) {
+            return PoolCandidate::scopeWhereHasPoolCandidateCommunity($query, $communityId);
+        });
+    }
+
+    public function whereHasDiploma(?bool $hasDiploma): self
+    {
+        if ($hasDiploma) {
+            return $this->where('has_diploma', true);
+        }
+
+        return $this;
+    }
+
+    public function wherePositionDurationIn(?array $positionDuration): self
+    {
+        if (empty($positionDuration)) {
+            return $this;
+        }
+
+        return $this->where(function ($query) use ($positionDuration) {
+            foreach ($positionDuration as $index => $duration) {
+                $query->orWhereJsonContains('position_duration', $duration);
+            }
+        });
+    }
+
+    public function whereEquityIn(?array $equity): self
+    {
+        if (empty($equity)) {
+            return $this;
+        }
+
+        // OR filter - first find out how many booleans are true, create array of all true equity booleans
+        // equity object has 4 keys with associated booleans
+        $equityVars = [];
+        if (array_key_exists('is_woman', $equity) && $equity['is_woman']) {
+            array_push($equityVars, 'is_woman');
+        }
+        if (array_key_exists('has_disability', $equity) && $equity['has_disability']) {
+            array_push($equityVars, 'has_disability');
+        }
+        if (array_key_exists('is_indigenous', $equity) && $equity['is_indigenous']) {
+            array_push($equityVars, 'is_indigenous');
+        }
+        if (array_key_exists('is_visible_minority', $equity) && $equity['is_visible_minority']) {
+            array_push($equityVars, 'is_visible_minority');
+        }
+
+        // 3 fields are booleans, one is a jsonb field
+        return $this->where(function ($query) use ($equityVars) {
+            foreach ($equityVars as $index => $equityInstance) {
+                if ($equityInstance === 'is_indigenous') {
+                    $query->orWhereJsonLength('indigenous_communities', '>', 0);
+                } else {
+                    $query->orWhere($equityVars[$index], true);
+                }
+            }
+        });
+    }
+
+    public function whereNameAndEmailNotIn(?array $negationArray): self
+    {
+        if (isset($negationArray) && count($negationArray) > 0) {
+            foreach ($negationArray as $value) {
+                $this->whereNot('first_name', 'ilike', $value);
+                $this->whereNot('last_name', 'ilike', $value);
+                $this->whereNot('email', 'ilike', $value);
+            }
+        }
+
+        return $this;
+    }
+
+    public function publicProfileSearch(?string $search): self
+    {
```

Comment:
Should this not have that "where" prefixing?

Reply:
Good catch, now lets hope my change did not break naything :laughing: 

[add prefix for public profile search](https://github.com/GCTC-NTGC/gc-digital-talent/pull/13449/commits/b9db5b579b8963c1cf9518df5b6c86268292c7b5)